hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/processor/ServiceContextImpl.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright (c) 2008-2019, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.jet.impl.processor;
+
+import com.hazelcast.jet.JetInstance;
+import com.hazelcast.jet.core.Processor;
+import com.hazelcast.jet.core.ProcessorSupplier;
+import com.hazelcast.jet.pipeline.ServiceFactory;
+import com.hazelcast.jet.pipeline.ServiceFactory.ServiceContext;
+import com.hazelcast.logging.ILogger;
+
+import javax.annotation.Nonnull;
+
+public class ServiceContextImpl implements ServiceContext {
+    private final int memberCount;
+    private final int memberIndex;
+    private final int localIndex;
+    private final boolean hasLocalSharing;
+    private final boolean hasOrderedAsyncResponses;
+    private final int maxPendingCallsPerProcessor;
+    private final String vertexName;
+    private final ILogger logger;
+    private final JetInstance jetInstance;
+
+    public ServiceContextImpl(
+            @Nonnull ServiceFactory<?> serviceFactory,
+            @Nonnull ProcessorSupplier.Context context
+    ) {
+        this.memberCount = context.memberCount();
+        this.memberIndex = context.memberIndex();
+        this.localIndex = context instanceof Processor.Context
```

Comment:
It's a bit awkward that we use two styles of checking the same thing (that the service is shared). I'd prefer to first establish `hasLocalSharing` and then assert the type of the context, which must be `Processor.Context` if there is no local sharing and vice versa.